### PR TITLE
fix(taskframework): dlm task show log not exists

### PR DESCRIPTION
**Pull request by krihy**
#### What type of PR is this?

type-bug
module-taskframework

#### What this PR does / why we need it:
1. throw exception when  destroy process failed
2. do final work before stop task 
3. get log by forward to remote odc server with try catch, avoid remote server is not reachable

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/BaseTask.java [hunk 1]
```diff
@@ -111,6 +109,18 @@ private void initCloudObjectStorageService() {
         storageConfig.ifPresent(osc -> this.cloudObjectStorageService = CloudObjectStorageServiceBuilder.build(osc));
     }
 
+    private void close() {
+        if (closed.compareAndSet(false, true)) {
```
[the remark this answers is not in the document]
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/BaseTask.java:113]: ok, this only node can stop this task at one time

**Review by guowl3 — DISMISSED** [review]
LGTM

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/schedule/daemon/CheckRunningJob.java [hunk 2]
```diff
@@ -72,30 +72,16 @@ private void handleJobRetryingOrFailed(JobEntity jobEntity) {
...
+        if (isNeedRetry) {
             log.info("Need to restart job, destroy old executor completed, jobId={}.", a.getId());
             int rows = getConfiguration().getTaskFrameworkService()
-                    .updateStatusDescriptionByIdOldStatusAndExecutorDestroyed(a.getId(), JobStatus.RUNNING,
+                    .updateStatusDescriptionByIdOldStatus(a.getId(), JobStatus.RUNNING,
                             JobStatus.RETRYING, "Heart timeout and retrying job");
             if (rows > 0) {
                 log.info("Job {} set status to RETRYING.", a.getId());
+            } else {
+                throw new TaskRuntimeException("Set job status to FAILED failed, jobId=" + jobEntity.getId());
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/schedule/daemon/CheckRunningJob.java:84]: FAILED --> RETRYING ?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/BaseTask.java [hunk 3]
```diff
@@ -111,6 +109,18 @@ private void initCloudObjectStorageService() {
...
+            try {
+                doClose();
+            } catch (Throwable e) {
+                // do nothing
+            }
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/BaseTask.java:118]: Task completed

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/schedule/daemon/CheckRunningJob.java (hunk 2 above)
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/schedule/daemon/CheckRunningJob.java]: yes, that's right , i fix it

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/BaseTask.java (hunk 3 above)
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/BaseTask.java:118]: done

**Review by yizhouxw — APPROVED** [review]

**Pull request merged**